### ISSUE TITLE
Implement GO batch delimiter in SQL Server

### ISSUE THREAD

**Issue by kAlvaro** [HeidiSQL/HeidiSQL#85]
It'd be nice that HeidiSQL implements the  [GO command ](https://docs.microsoft.com/en-us/sql/t-sql/language-elements/sql-server-utilities-statements-go) provided by Microsoft clients (SQLCMD, SQL Server Management Studio, etc.).

Please note that it is neither a T-SQL statement nor a statement delimiter (that'd be `;`). It's a client-side batch delimiter that has the effect of submitting a group of one or more statements to the server in a single request. For this reason, the *Set delimiter used in SQL execution* feature isn't a workaround:

- Being a command, `GO`can't just appear anywhere
- You can also have (and should) regular delimiters

Currently, HeidiSQL submits it to the server and gets back a syntax error:

> SQL Error (102): Incorrect syntax near 'go'.

Ideally, it should group together the appropriate statements and submit then at once. In e.g. large insertions this boosts performance.

To consider: would there be a point for yet another toolbar button option to disable it? There're circumstances where you may want to submit statements one by one, e.g. in a very large file that doesn't have `GO` commands.

**Comment by ansgarbecker** [comment]
Another question is: should `GO` be limited to SQL Servers in HeidiSQL? I guess there is no point why HeidiSQL should disable that on MySQL or PG?

**Comment by kAlvaro** [comment]
Perhaps *Send batch in one go* would need to be renamed to *Send file in one batch* but, yeah, it's unclear how it should interact with `GO` :-?

About MySQL or PostgreSQL... A key point is that we may would end up with scripts that would only run in HeidiSQL—probably not desirable (not ready to dominate the world yet). But to be fair, I've just learnt that HeidiSQL happily ignores `delimiter //` under SQL Server, possibly for years, and nothing broke.

**Comment by reubendevries** [comment]
Yeah as far as I know, MySQL and Postgres scripts don't use the GO delimiter, so I wouldn't recommend that you allow it.

**Comment by sanderd17** [comment]
Parsing "GO" should be pretty straightforward. It's not allowed together with other statements.

> A Transact-SQL statement cannot occupy the same line as a GO command. However, the line can contain comments. 
(https://docs.microsoft.com/en-us/sql/t-sql/language-elements/sql-server-utilities-statements-go)

In my own scripting tools (just some python scripts), I've also implemented a "GO" delimiter, where I just split the file on "\nGO". Though that's not completely valid, as it doesn't take comments into account, and GO doesn't need to be the first thing on the line either.

**Comment by wilfz** [comment]
This feature would help users of MS-SQL Server a lot.
To my knowledge it ist is then only way to create functions and procedures in larger SQL scripts.

**Comment by ansgarbecker** [comment]
`GO` is not a SQL keyword but a client‑side batch separator specific to Microsoft SQL Server tools (SSMS, sqlcmd, etc.). HeidiSQL is a multi‑DB client (MySQL/MariaDB, PostgreSQL, SQL Server, SQLite etc.) and I try to avoid implementing vendor‑specific scripting syntax that only exists in one tooling ecosystem.

Supporting `GO` would require a custom pre‑parser, special‑casing edge cases (e.g. GO in comments or strings), and defining non‑standard behavior for all other database backends where GO has no meaning. For that reason I don’t plan to add GO support in HeidiSQL. If you have scripts that use GO, they should either be run with SQL Server tools or pre‑processed to remove/convert GO before running them in HeidiSQL.
